packages/charts/src/components/ScatterChart/ScatterChart.tsx

```diff
@@ -0,0 +1,242 @@
+}
+
+interface ScatterDataObject {
+  label?: string;
+  data?: any[];
+  color?: CSSProperties['color'];
+}
+
+export interface ScatterChartProps extends IChartBaseProps {
+  dataset?: ScatterDataObject[];
+  /**
+   * An array of config objects. Each object is defining one line in the chart.
+   *
+   * <h4>Required properties</h4>
+   * - `accessor`: string containing the path to the dataset key this line should display. Supports object structures by using <code>'parent.child'</code>.
+   *   Can also be a getter.
+   *
+   * <h4>Optional properties</h4>
+   *
+   * - `label`: Label to display in legends or tooltips. Falls back to the <code>accessor</code> if not present.
+   * - `color`: any valid CSS Color or CSS Variable. Defaults to the `sapChart_Ordinal` colors
+   * - `formatter`: function will be called for each data label and allows you to format it according to your needs
+   * - `hideDataLabel`: flag whether the data labels should be hidden in the chart for this line.
+   * - `DataLabel`: a custom component to be used for the data label
+   * - `width`: line width, defaults to `1`
+   * - `opacity`: line opacity, defaults to `1`
+   *
+   */
+  measures?: MeasureConfig[];
+}
+
+const measureDefaults = {
+  formatter: defaultFormatter
+};
+
+/**
+ * <code>import { LineChart } from '@ui5/webcomponents-react-charts/lib/LineChart';</code>
+ */
+const ScatterChart: FC<ScatterChartProps> = forwardRef((props: ScatterChartProps, ref: Ref<any>) => {
+  const {
+    dataset,
+    loading,
+    noLegend = false,
+    noAnimation = false,
+    onDataPointClick,
+    onLegendClick,
+    style,
+    className,
+    tooltip,
+    slot
+  } = props;
+
+  const chartConfig = useMemo(() => {
+    return {
+      yAxisVisible: false,
+      xAxisVisible: true,
+      gridStroke: ThemingParameters.sapList_BorderColor,
+      gridHorizontal: true,
+      gridVertical: false,
+      legendPosition: 'bottom',
+      legendHorizontalAlign: 'left',
+      zoomingTool: false,
+      resizeDebounce: 250,
+      ...props.chartConfig
+    };
+  }, [props.chartConfig]);
+
+  const { measures } = usePrepareDimensionsAndMeasures([], props.measures, {}, measureDefaults);
+
+  const tooltipValueFormatter = useTooltipFormatter(measures);
+  const chartRef = useConsolidatedRef<any>(ref);
+  const onItemLegendClick = useLegendItemClick(onLegendClick);
+
+  const onDataPointClickInternal = useCallback(
+    (payload, eventOrIndex) => {
+      if (eventOrIndex.dataKey && onDataPointClick) {
+        onDataPointClick(
+          enrichEventWithDetails(
+            {},
+            {
+              value: eventOrIndex.value,
+              dataKey: eventOrIndex.dataKey,
+              dataIndex: eventOrIndex.index,
+              payload: eventOrIndex.payload
+            }
+          )
+        );
+      }
+    },
+    [onDataPointClick]
+  );
+
+  const isBigDataSet = dataset?.length > 30 ?? false;
+
+  const xMeasure = measures.filter(({ axis }) => axis === 'x');
```

Comment:
Why not using `.find` instead of filter? this will help you to remove all access with `?.[0]` in the code below